src/imperial.md

```diff
@@ -0,0 +1,101 @@
+---
+layout: base
+title: Documentation
+description: 
+---
+
+## Imperial Service
+
+... (To be written)
+
+### Serial packet format
+
+| Byte   | Purpose                         |
+| ------ | ------------------------------- |
+| 1-8    | Header will equal `IMPERIAL`    |
+| 9      | Type (see below)                |
+| 10     | Length                          |
+| 11...X | Data                            |
+| X + 11 | Checksum (Not for RPC/Response) |
+
+
+### Current State (_Device to client_)
+
+Current state type: `0x01`
+
+The current status of the provisioning service and will write and notify any listening clients for instant feedback.
+
+| Value  | State                  | Purpose                                          |
+| ------ | ---------------------- | ------------------------------------------------ |
+| `0x02` | Ready (Authorized)     | Ready to accept credentials.                     |
+| `0x03` | Provisioning           | Credentials received, attempt to connect.        |
+| `0x04` | Provisioned            | Connection successful.                           |
```

Comment:
When it's this state, we should allow the device to specify a url that the client can let the user visit to interact with the device, similar to the URL returned from "Send Wi-Fi settings"